event_creation/submission/parsers/elemem_parsers.py

```diff
@@ -107,6 +108,444 @@ def event_stimulation(self, evdata):
+            ann_outputs = self._parse_ann_file(str(self._trial) + '-' + " ".join(store_name.lower().split('_')))
+
+        except:
+            ann_outputs = []
+            print(("MISSING ANNOTATIONS FOR %s" % rec_start_event.store))
+
+        for recall in ann_outputs:
+            new_event = self._new_rec_event(recall, rec_start_event)
+            # new_event = self._identify_intrusion(events, new_event)
+
+            evtype = 'CUED_REC_WORD_VV' if "<>" in new_event["item"] else 'CUED_REC_WORD'
+            new_event.type = evtype
+
+            # TODO: this matches previous events, but without this CUED_REC events lack any annotation
+            # if new_event.intrusion > 0:
+            #     events[(events["type"] == 'WORD') & (events["item"] == new_event["item"])].intruded = 1
+            # elif new_event.intrusion == 0:
+            #     events[(events["type"] == 'WORD') & (events["item"] == new_event["item"])].recalled = 1
+
+            new_event["intrusion"] = -999
+
+            events = np.append(events, new_event).view(np.recarray) 
+
+        return events
+
+    def modify_store_recall(self, events):
+        rec_start_event = events[-1]
+        rec_start_time = rec_start_event.mstime
+        ann_outputs = self._parse_ann_file("final store-0")
+
+        for recall in ann_outputs:
+            new_event = self._new_rec_event(recall, rec_start_event)
+            evtype = 'SR_REC_WORD_VV' if "<>" in new_event["item"] else 'SR_REC_WORD'
+            new_event.type = evtype
+            new_event.trial = -999 # to match old events
+            new_event.intrusion = 0 if new_event["item"] in ["_".join(s.upper().split()) for s in self.STORES] else -1
+
+            events = np.append(events, new_event).view(np.recarray) 
+
+        return events
+
+    def modify_free_recall(self, events):
+        rec_start_event = events[-1]
+        rec_start_time = rec_start_event.mstime
+        try:
+            ann_outputs = self._parse_ann_file("final recall-0")
+        except:
+            ann_outputs = self._parse_ann_file("final free-0")
+
+        words = events[events["type"] == 'WORD']
+
+        for recall in ann_outputs:
+            new_event = self._new_rec_event(recall, rec_start_event)
+
+            # Create a new event for the recall
+            evtype = 'FFR_REC_WORD_VV' if "<>" in new_event["item"] else 'FFR_REC_WORD'
+            new_event.type = evtype
+            new_event.trial = -999 # to match old events
+
+            if new_event.intrusion >= 0:
+                new_event.intrusion = 0
+                events.finalrecalled[(events["type"] == "WORD") & (events["item"] == new_event["item"])] = 1
+            
+            events = np.append(events, new_event).view(np.recarray) 
+
+        return events
+
+    # LC: update trial and stimtag info (3,8HZ)
+    def modify_event_stimulation(self, events):
+        events[-1].trial = self._trial
+        events[-1].stim_params.burst_freq = int(self._stimtag[0])
```

Comment:
This appears inappropriate in structure.  The stimtag is a string which is coincidentally and for human readability convenience "3Hz" and "8Hz", but is only a string key to the stim configuration which contains the actual burst_freq which is used.  This lookup process should be replicated here to avoid future disasters of grabbing a character here of a text label, or grabbing e.g. "1" for "16Hz" or other issues.

Reply:
Resolved. 
I am grabbing the stim tag value, and using that to access the burst frequency information in the experimental config file. So stim tag doesn't necessarily have to be an integer value now.